core/templates/dev/head/pages/email_dashboard/EmailDashboard.js

```diff
@@ -0,0 +1,168 @@
+// Copyright 2016 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Controller and factory for oppia email dashboard page.
+ */
+
+oppia.factory('EmailDashboardDataService', [
+  '$http', function($http) {
+    var QUERY_DATA_URL = '/emaildashboarddatahandler';
+    var QUERY_STATUS_CHECK_URL = '/querystatuscheck';
+
+    // No. of query results to display on a single page.
+    var QUERIES_PER_PAGE = 10;
+    // Store latest cursor value for fetching next query page.
+    var latestCursor = null;
+    // Cache all fetched queries.
+    var queries = [];
+    // Index of currently showing page of query results.
+    var currentPageIndex = -1;
+
+    var fetchQueries = function() {
+      return $http.get(QUERY_DATA_URL, {
+        params: {
+          cursor: latestCursor
+        }
+      }).then(function(response) {
+        return response.data;
+      });
+    };
+
+    return {
+      submitQuery: function(data, callback) {
+        $http.post(QUERY_DATA_URL, {
+          data: data
+        });
+
+        var startQueryIndex = (currentPageIndex) * QUERIES_PER_PAGE;
+        var endQueryIndex = (currentPageIndex + 1) * QUERIES_PER_PAGE;
+        fetchQueries().then(function(data) {
```

Comment:
Yes, exactly. I was wondering why we need the server to control it at all, since the notion of a page is really entirely in the frontend. Perhaps the frontend should just pass it in?